.evergreen/mongodl.py

```diff
@@ -0,0 +1,936 @@
+    # Extract the "VERSION_ID" field
+    ver_id_re = re.compile(r'VERSION_ID=("?)(.*?)\1')
+    mat = ver_id_re.search(os_rel)
+    assert mat, 'Unable to detect VERSION_ID from [/etc/os-release] content:\n{}'.format(
+        os_rel)
+    ver_id = mat.group(2)
+    # Map the ID to the download ID
+    mapped_id = DISTRO_ID_MAP.get(os_id)
+    if mapped_id:
+        ver_mapper = DISTRO_VERSION_MAP.get(os_id)
+        if ver_mapper:
+            mapped_version = ver_mapper[ver_id]
+            ver_id = mapped_version
+        os_id = mapped_id
+    os_id = os_id.lower()
+    if os_id not in DISTRO_ID_TO_TARGET:
+        raise RuntimeError("We don't know how to map '{}' to a distribution "
+                           "download target. Please contribute!".format(os_id))
+    # Find the download target based on a filename-style pattern:
+    ver_table = DISTRO_ID_TO_TARGET[os_id]
+    for pattern, target in ver_table.items():
+        if fnmatch(ver_id, pattern):
+            return target
+    raise RuntimeError(
+        "We don't know how to map '{}' version '{}' to a distribution "
+        "download target. Please contribute!".format(os_id, ver_id))
+
+
+def user_caches_root() -> Path:
+    """
+    Obtain the directory for user-local caches
+    """
+    if sys.platform == 'win32':
+        return Path(os.environ['LocalAppData'])
+    if sys.platform == 'darwin':
+        return Path(os.environ['HOME'] + '/Library/Caches')
+    xdg_cache = os.getenv('XDG_CACHE_HOME')
+    if xdg_cache:
+        return Path(xdg_cache)
+    return Path(os.environ['HOME'] + '/.cache')
+
+
+def default_cache_dir() -> Path:
+    """
+    Get the path to the default directory of mongodl caches.
+    """
+    return user_caches_root().joinpath('mongodl').absolute()
+
+
+if TYPE_CHECKING:
+    DownloadResult = NamedTuple('DownloadResult', [('is_changed', bool),
+                                                   ('path', Path)])
+    DownloadableComponent = NamedTuple('DownloadableComponent', [
+        ('version', str),
+        ('target', str),
+        ('arch', str),
+        ('edition', str),
+        ('key', str),
+        ('data_json', str),
+    ])
+else:
+    DownloadResult = namedtuple('DownloadResult', ['is_changed', 'path'])
+    DownloadableComponent = namedtuple(
+        'DownloadableComponent',
+        ['version', 'target', 'arch', 'edition', 'key', 'data_json'])
+
+#: Regular expressoin that matches the version numers from 'full.json'
+VERSION_RE = re.compile(r'(\d+)\.(\d+)\.(\d+)(?:-([a-z]+)(\d+))?')
+STABLE_MAX_RC = 9999
+
+
+def version_tup(version: str) -> 'tuple[int, int, int, int, int]':
+    mat = VERSION_RE.match(version)
+    assert mat, ('Failed to parse "{}" as a version number'.format(version))
+    major, minor, patch, tag, tagnum = list(mat.groups())
+    if tag is None:
+        # No rc tag is greater than an equal base version with any rc tag
+        tag = STABLE_MAX_RC
+        tagnum = 0
+    else:
+        tag = {
```

Comment:
IIUC this currently considers `rc < alpha < beta` for getting the latest release. I think that may need to be flipped. `rc` is the most stable.